lib/Scheduling/TestPasses.cpp

```diff
@@ -83,13 +83,11 @@ static void constructProblem(Problem &prob, FuncOp func) {
   // of integer attributes
   if (auto attr = func->getAttrOfType<ArrayAttr>("auxdeps")) {
     auto &ops = prob.getOperations();
-    unsigned nOps = ops.size();
-    for (auto &elemArr : parseArrayOfArrays(attr)) {
+    [[maybe_unused]] unsigned nOps = ops.size();
+    for ([[maybe_unused]] auto &elemArr : parseArrayOfArrays(attr)) {
       assert(elemArr.size() >= 2 && elemArr[0] < nOps && elemArr[1] < nOps);
-      Operation *from = ops[elemArr[0]];
-      Operation *to = ops[elemArr[1]];
-      auto res = prob.insertDependence(std::make_pair(from, to));
-      assert(succeeded(res));
+      assert(succeeded(prob.insertDependence(
```

Comment:
is it safe to move this code into the assert here?

Reply:
In general no to both of these comments. You don't want to do anything with side effects inside an assert, because those side effects won't happen in a release build. The pattern we've used for things like this is:

```
auto res = prob.insertDependence(...);
assert(succeeded(res));
(void)res;
```

Reply:
Thanks, this will rise **ninja check-circt** get errors , sorry to not check it carefully.
 I will to check it both in RELEASE and DEBUG type and  later to submit a commit.